app/extend/context.js

```diff
@@ -102,15 +102,14 @@ module.exports = {
 
     if (useSession) {
       this.session[sessionName] = secret;
-    } else {
-      const cookieOpts = {
-        domain: cookieDomain && cookieDomain(this),
-        signed: false,
-        httpOnly: false,
-        overwrite: true,
-      };
-      this.cookies.set(cookieName, secret, cookieOpts);
     }
+    const cookieOpts = {
```

Comment:
这个不能是默认行为，因为默认打开是不安全的。需要改成配置的。

Reply:
useSessionWithCookie: false,          // Only work if useSession is true. It will set Cookie cookieName with csrf token. csrf token compare with cookie and session.

Reply:
为什么不直接用 cookie？

Reply:
> 为什么不直接用 cookie？

直接用cookie中的ctoken，目前的校验是http header的x-csrf-token的cookie double check。如果攻击者模仿cookie的ctoken和header的x-csrf-token保持同一个值，就可以进行攻击。
另一个思路是对token算法校验，不过我们这样修改成本更低。

Reply:
如钉钉沟通，加上 useSessionWithCookie 之后，cookie 中也会含有 token，这和直接存 cookie 中没有区别（只是客户端无法修改）。而 csrf 防范方案只针对跨站攻击的防范，通过非浏览器请求的方案是不在考虑范围之内（也无法防范的）。所以不建议增加这个配置项，对跨站防范没有帮助。